File: php/CreateCollectionpoint.php
<?php
/**
 * sample of create an collection point waybill,
 *
 */
include_once "../EncryptUtil.php";
include_once "../HttpRequestUtil.php";

$productInfo = [
    [   "name"=>"Iphone 7 golden color",//name of the product
        "serial_number"=>"123456",//serial number of the product,
        "qty"=>1, //quantity of the product
        "price"=>2599, //price of the product
        "volume"=>"23 x 12 x 12" //shop id of the customer choose,Length x Width x Height
    ],
    [
        "name"=>"T shirt nike",//name of the product
        "serial_number"=>"123243",//serial number of the product,
        "qty"=>1, //quantity of the product
        "price"=>23,//price of the product
        "volume"=>"23 x 12 x 12" //shop id of the customer choose,Length x Width x Height
    ]
];
$requestTime = time(); //timestamp of request
$secretKey = ""; //secretKey from Fodel
$params = array();
$params["recipient_name"] = "Fiona Williams";//full name of customer
$params["phone"] = "0508873269";//phone of the customer
$params["address"] = "Dubai Opera,Dubai,United Arab Emirates";//address of the customer
$params["weight"] = 420;//weight of the parcel,unit: g
$params["is_cod"] = 1;//is the cod or not : 1 yes,0:no
$params["price_cod"] = 1000;//amount of the cod,if it is not cod ,put 0,
$params["order_no"] = "";//order number
$params["app_key"] = "";//app key of the fodel
$params["shop_id"] = 200001;//shop id of the customer choose
$params["ts"] = $requestTime;
$params["product_info"] = json_encode($productInfo); //product information , just put the json format
//generate the sign from the paramters with eht secretKey
$signString = EncryptUtil::generateSign($params,$secretKey);
$params['sign'] = $signString;

//Set header ,add the api version to the parameter fodel_api_version
$header = array();
$header[]="fodel_api_version: v1";
$httprequest = new HttpRequestUtil();
$httprequest->setBaseUrl("http://api.test.fo-del.com");
$respone = $httprequest->post("/parcel/create",$params,true,$header);
echo $respone->body;

File: java/CreateCollectionpoint.java
import com.alibaba.fastjson.JSONException;
import com.alibaba.fastjson.JSONObject;

import java.io.IOException;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class CreateCollectionpoint {
	public static void main(String[] args) throws JSONException, IOException, ParseException, FodelAPIException{
		
		String secretKey = "";//secretKey from Fodel
		String requestTime = (System.currentTimeMillis()/1000)+"";
		Map<String ,String> params = new HashMap<String,String>();
		params.put("recipient_name", "EDer Uber");//full name of customer
		params.put("phone", "971522545710");//phone of the customer
		params.put("weight", "2000");//weight of the parcel,unit: g
		params.put("is_cod", "1");//is the cod or not : 1 yes,0:no
		params.put("price_cod", "1000");//amount of the cod,if it is not cod ,put 0,
		params.put("order_no", "2399453401123");//order number
		params.put("app_key", ""); //app key from fodel
		params.put("shop_id", "36"); //shop_id ,from the map of the
		params.put("ts", requestTime);
		List<ReturnGoodList> returnGoodLists = new ArrayList<>();
		returnGoodLists.add(new ReturnGoodList(Byte.parseByte("2"),"Iphone 7","123456","12.5","12 x 12 x 34"));
		params.put("product_info",JSONObject.toJSONString(returnGoodLists));
		String signString = EncryptUtil.generateSign(params, secretKey);
		params.put("sign", signString);
		String requestBody = JSONObject.toJSONString(params);
		JSONObject jsonObject = HttpRequestUtil.request("POST", "http://api.test.fo-del.com/shipment/create", requestBody);
		System.out.println(jsonObject);	
	}
}
